packages/nitro-protocol/contracts/MultiAssetHolder.sol

```diff
@@ -189,33 +190,43 @@ contract MultiAssetHolder is IMultiAssetHolder, ForceMove {
                 targetOutcomeBytes,
                 (Outcome.OutcomeItem[])
             );
-            Outcome.AssetOutcome memory assetOutcome = abi.decode(
-                outcome[assetIndex].assetOutcomeBytes,
-                (Outcome.AssetOutcome)
-            );
-            require(
-                assetOutcome.assetOutcomeType == Outcome.AssetOutcomeType.Allocation,
-                '!allocation'
-            );
-            allocation = abi.decode(
-                assetOutcome.allocationOrGuaranteeBytes,
-                (Outcome.AllocationItem[])
-            );
+            {
+                Outcome.AssetOutcome memory assetOutcome = abi.decode(
+                    outcome[assetIndex].assetOutcomeBytes,
+                    (Outcome.AssetOutcome)
+                );
+                require(
+                    assetOutcome.assetOutcomeType == Outcome.AssetOutcomeType.Allocation,
+                    '!allocation'
+                );
+                allocation = abi.decode(
+                    assetOutcome.allocationOrGuaranteeBytes,
+                    (Outcome.AllocationItem[])
+                );
+            }
```

Comment:
```suggestion
            Outcome.AssetOutcome memory assetOutcome = abi.decode(
                outcome[assetIndex].assetOutcomeBytes,
                (Outcome.AssetOutcome)
            );
            require(
                assetOutcome.assetOutcomeType == Outcome.AssetOutcomeType.Allocation,
                '!allocation'
            );
            allocation = abi.decode(
                assetOutcome.allocationOrGuaranteeBytes,
                (Outcome.AllocationItem[])
            );
```

As far as I can tell this extra indent + [block scope](revert-to-old-event-system) is unneeded and probably a mistake.

Reply:
I took your word for it and committed this suggestion along with the other one. However, it caused a "stack too deep" compiler error. I don't think the block scoping here is terribly neat or optimal, but it does seem to avoid that problem. If you want to play around with it some more, maybe have a go on a new branch and check each change compiles successfully.